packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -24,19 +38,16 @@ class _FlatButtonState extends MaterialButtonState<FlatButton> {
 
   int get elevation => 0;
 
-  Color getColor(BuildContext context, { bool highlight }) {
-    if (!config.enabled || !highlight)
-      return null;
-    switch (Theme.of(context).brightness) {
-      case ThemeBrightness.light:
-        return Colors.grey[400];
-      case ThemeBrightness.dark:
-        return Colors.grey[200];
-    }
+  Color getColor(BuildContext context) {
+    if (!config.enabled)
+      return config.disabledColor ?? null;
```

Comment:
The `?? null` seems pointless because the right half is only ever used if the left half is null already.